### ISSUE TITLE
[Feature Request] Repeating tasks

### ISSUE THREAD

**Issue by stoopidJSON** [nextcloud/tasks#34]
Repeating tasks, like calendar items, would be a really useful feature.  Right now I'm putting repeating tasks in the calendar (stuff like taking out the trash weekly) but really that'd be better as a repeating task in this app.

I'm not very familiar with the Nextcloud platform, but it looks pretty straight forward.  I will fork this repo and try my hand at adding this feature myself.  Between my unfamiliarity with the platform, and other projects, I may not ever get it done though.  I'm posting the idea here in hopes that someone else who is more knowledgable about this project take up this cause.

**Comment by jurgenhaas** [comment]
I second that, recurring tasks would be really useful because I could then replace my GTD tool and bring everything into NextCloud tasks,

**Comment by revjtanton** [comment]
I've looked at doing this myself a little bit, and I think I have an idea on how this could function in-line with current Tasks functionality.

Currently a Task can span many days or months.  The has a progress bar.  This is not similar to other task programs like Todoist or others where a task is simply due on a date, or recurring.  

To create a repeating option for Nextcloud Tasks I suggest it act as a reminder for a task in progress.  So you set your task start date/end date, then under a repeating or reminders tab (just like a calendar item) you set how you will be reminded (pop-up, email, etc) and then the interval (every 1 day, 1 month, etc--this is where it is different than Calendar and is a combination of the reminder and recurring tabs).  I'm not sure how this would work with third party items like DAVdroid/Tasks on Android (for example), but I think this type of recurring reminder would be complementary to the way Tasks is currently implemented in Nextcloud.

**Comment by Perelandra0x309** [comment]
+1 for repeating tasks

**Comment by ddanielww** [comment]
I would also like this feature

**Comment by xpilgrim** [comment]
me also ;-)

**Comment by schnups** [comment]
yes, this would be awesome

**Comment by durchflieger** [comment]
I like it too

**Comment by tschiu** [comment]
This would be very helpful.
Maybe a quick method would be to put a database table in place for repeating task info, then install a cron job to read that table and create the new entries in the app database table?

**Comment by rnolty** [comment]
+1

**Comment by serenitatis** [comment]
+100000

**Comment by Trollwut** [comment]
Hello! Are there any plans on implementing this feature? A deadline?

Love the whole software and this is the only feature that keeps me using another app for managing my shit. :)

**Comment by SeaDude** [comment]
I would love to see this feature as well. This would make things much easier for my family to track our activities. I'm willing to give $20 toward this feature if it would help.

**Comment by Trollwut** [comment]
@SeaDude at Bountysource you can pay a bounty for specific issues, so that developers get paid if they solve them.

Haven't used that platform for specific bounties (just to donate devs in general) and don't know if there's an open issues for repeating tasks. But I guess that would be where those 20 bucks could be heading to. :)

**Comment by rsudev** [comment]
An implementation of recurring tasks I had used before (in another app) just spawned the next occurrence, if the current one was marked as done. Currently I do this in a more manual fashion by just moving the entry to the next occurrence instead of recording it as done.
Not sure if the implementation suggestion above provides the same level of information (see the next occurrence as a due entry in the task list).
Good interworking with clients (DAVdroid/Tasks and Thunderbird/Tasks in my case) is of course quite crucial...
And of course a big thanks for providing this app in the first place!

**Comment by MartinConsultingServicesInc** [comment]
I am also in favor of repeating tasks.  I wanted to add a few more thoughts.

Suppose I schedule a repeating task for every Monday.  On Monday, when I look at my "Today" tasks, it should appear.  If I mark it off as complete, it should disappear until next Monday, when it should reappear.  I suspect there's wide consensus on that point.  

If I don't do it on Monday, however, to my mind it's still outstanding, so on Tuesday it should still appear as an (over-)due task.  There may be a few cases where people will want it to just disappear (e.g. if you didn't take out the garbage on Monday, there's no point in doing it on Tuesday), but those are probably few enough that just checking them off as done is sufficient.  Perhaps an option is needed there, but I think it's probably low priority.

For tasks that are completed late, there's the question of when to reschedule them.  In many cases, if it is scheduled for Monday, it should be rescheduled for next Monday regardless of when it has been completed.  However, there are other cases where the interval is more important then the start date.  Suppose I want to change the oil in my car every three months.  I set up a task to start on January 1 and repeat every 3 months.  Come April, though, I'm out of town and don't change the oil until April 15th.  Some people will think the next oil change should be June 1, but others will want June 15 (because that's when the oil is 3 months old).  Other scheduling apps handle this by allowing rescheduling to be on a date (1st of the month every 3rd months) or on an interval (every 3 months).

Anything that can be done for repeating tasks would be wonderful.  The above is just to help raise the issues a developer may want to consider.
  If I complete the task on Tuesday, it should disappear until

**Comment by DJCrashdummy** [comment]
just a hint if some of you didn't know:
(i had problems importing an calendar with recurring tasks and alarms as you can see in https://github.com/nextcloud/calendar/issues/777, but) syncing recurring tasks, ticking and modifying them in a client seem to work fine... but it's really a pity, that only the first task is displayed in nextcloud-tasks. - the only way for now is to don't tick these "master", because this will tick all it's "subtasks" in your clients.
...and to be on the safe side i also don't edit them within nextcloud to don't mess things up. :disappointed:

**Comment by Dr-Cool** [comment]
Just posted a $10 bounty at Bountysource for this feature. Will accept the claim if implementation is compatible with repeating events in [Tasks (Astrid To-Do List Clone)](https://tasks.org/).

**Comment by SkylightXD** [comment]
+1

**Comment by randyheydon** [comment]
@MartinConsultingServicesInc I've been using [etm](http://people.duke.edu/~dgraham/etmtk/) up until now, and it handles what you describe in a simple way.  [Repeating tasks](http://people.duke.edu/~dgraham/etmtk/UserManual.html#r-repetition-rule) can have an "overdue" rule to select desired behaviour.  It defaults to "keep", or can be set to "skip" or "restart".  "Keep" has overdue tasks stay until complete, and future repetitions are not affected; "skip" has overdue tasks disappear, and future repetitions are not affected (like your garbage example); and "restart" has overdue tasks stay until complete, and further repetitions are scheduled based on the completion date (like your oil change example).

In my experience, these three options are sufficient to cover most any case I've encountered, and might be good inspiration for repeating tasks in Nextcloud.

**Comment by 995190** [comment]
Any update on this? It's the only thing currently missing that disables me from completely switching to Tasks with my work management :)

**Comment by crazybadger** [comment]
I was just about to add a feature request for this until I found this!  Only just started playing with this in Nextcloud, but so far it looks really good.  This was the first thing I found to be missing, as a long time OmniFocus user.

**Comment by mrstegeman** [comment]
This is actually already supported by the server. If you manage your task list with an external app, such as the Tasks app in macOS or Evolution on Linux, you can manage recurrence. There's just no way to do it through Nextcloud's UI.

**Comment by crazybadger** [comment]
What's the Tasks app in macOS?

**Comment by mrstegeman** [comment]
Sorry, it's called Reminders, not Tasks.

**Comment by crazybadger** [comment]
Ah, right.  I didn't even think about that or know that I could sync it with Nextcloud.  Unfortunately, recurring tasks is about all it's useful for as you can't nest tasks.  Should mean recurrence wouldn't be too difficult to add into Nextcloud though?

**Comment by thrdroom** [comment]
> 
> 
> Just posted a $10 bounty at Bountysource for this feature. Will accept the claim if implementation of repeating tasks is compatible with [Tasks (Astrid To-Do List Clone)](https://tasks.org/).

same goes for me. Just added 15$ to the bounty. I will accept the claim if implementation of repeating tasks is compatible with Thunderbird(Lightning) and [Tasks (Astrid To-Do List Clone)](https://tasks.org/)

**Comment by netzwerch** [comment]
> But I did not find any Android app which also supports this. DAVdroid syncs, but neither aCalendar+ nor OpenTasks can handle these recurring tasks. Are there any solutions? Thanks.

OpenTasks has it in its beta channel https://github.com/dmfs/opentasks/issues/462
And I can confirm that it works for the Android App "Tasks" when using CalDav https://tasks.org/ https://f-droid.org/packages/org.tasks/

**Comment by canufarm** [comment]
> And I can confirm that it works for the Android App "Tasks" when using CalDav https://tasks.org/ https://f-droid.org/packages/org.tasks/

Great, thanks! Tasks works. It's better than the aCalendar+ OpenTasks combination.

The only issues which remain are, task reminders set in Android/Tasks are not synced to eM Client (seems they in principle are not synced to Nextcloud) and when adding a new task on Android/Tasks after sync to eM Client the timezone is set to "Floating", no idea why.

**Comment by MisterY** [comment]
aCalendar+ with OpenTasks has always worked for me correctly. It allows me to see the tasks with due dates on the calendar. So, statements like "it is better than aCalendar OpenTasks combination" are dubious at best.

**Comment by thrdroom** [comment]
> So, statements like "it is better than aCalendar OpenTasks combination" are dubious at best.

Nope, not dubious at all.
Tasks.org app is:

1. Not in Beta-stadium
2. More frequently updated than opentasks
3. Don't needs 5 years to deliver recurring tasks!
4. Looks better, has nicer UI and UX
5. All in one app(no aCalendar(not in fdroid) app needed)
6. FOSS(no proprietary aCalendar app needed)

Therefore Tasks.org app is superior to opentasks. If you use it, please dont forget to donate to the devs ;)

**Comment by IpeTek** [comment]
I would still love to see recurrence in NC Tasks, as well. Thunderbird is working slowly on my PC, nowadays.

**Comment by WhoNeedszZz** [comment]
Just seeing this and was looking for the same feature. I am okay with the Tasks providing the recurrence input and then it being applied by the server, but what exactly is the issue with adding this option to the Nextcloud interface?

**Comment by vozeldr** [comment]
Would love to see this feature added.

**Comment by RavetcoFX** [comment]
This somewhat works for me, I can't modify the repeating on nextcloud, but when I set it too repeat in the android app Tasks.org

**Comment by TobiasDev** [comment]
> 
> 
> This somewhat works for me, I can't modify the repeating on nextcloud, but when I set it too repeat in the android app Tasks.org, it saves and syncs via Nextcloud

When you do that, what happens if you mark the task as done on the website? Does that stop the repeating?

**Comment by mrstegeman** [comment]
I tested this with the Reminders app on macOS. When marking as done on the website, it marks the entire task as completed and does not generate the repeated task.

**Comment by TobiasDev** [comment]
Yeah, this was the kind of behavior I was suspecting would happen. :) I mean, it still works, as long as you mark it in the Android app rather than in the Web view (or Gnome Todo). I just need to keep that in mind when I use it, since I very often have the Todo list open on the side when doing work.

**Comment by BeatLink** [comment]
Any updates on this?

**Comment by BeatLink** [comment]
Its a shame that its not implemented in the nextcloud interface. Recurrence in tasks isnt a bolt on non standard feature, its an official part of the ical standard https://icalendar.org/iCalendar-RFC-5545/3-8-5-3-recurrence-rule.html . I dont understand why one of the oldest and most welcomed features hasnt been worked on in  the UI yet.

**Comment by raimund-schluessler** [comment]
> Its a shame that its not implemented in the nextcloud interface. Recurrence in tasks isnt a bolt on non standard feature, its an official part of the ical standard https://icalendar.org/iCalendar-RFC-5545/3-8-5-3-recurrence-rule.html . I dont understand why one of the oldest and most welcomed features hasnt been worked on in the UI yet.

Please feel free to send a PR. And since I expect none: Complaining about a feature not being implemented won't gift me more time to work on the app. After all, this is a free-time hobby project.

**Comment by blacktemplar** [comment]
Any news on this feature and its interopability with Thunderbird Lightning Tasks and the android app Tasks?

**Comment by ivnvitx** [comment]
Is anybody working on this? It's been YEARS. And it is pretty basic stuff for tasks to have repeatability.

**Comment by raimund-schluessler** [comment]
> Is anybody working on this? It's been YEARS. And it is pretty basic stuff for tasks to have repeatability.

No, this is not been worked on at the moment. So please feel free to start working and provide a PR. :)

This is an OpenSource app developed by people in their freetime (mainly). Every development happens in the open, if someone would work on a feature, there would be a PR or at least a branch where you can track the progress. If there is nothing like this, please refrain from asking every few days for the progress of feature xy. It distracts and annoys people and makes it more unlikely that someone is in the mood to start implementing it.

**Comment by marbetschar** [comment]
@raimund-schluessler thank you for staying calm and gentle 🏆️ - your patience made my day! 

![Again? Seriously?](https://media.giphy.com/media/ka55CqnDNjQ7iIKtRa/giphy.gif)

**Comment by timkrief** [comment]
I started using the FLOSS android app tasks.org and it has repeating tasks. Changing a task with nextcloud tasks removes the repeating aspect of the task when syncing back to the app. So yes this would be really nice to have it here as well or at least in a first time, not to erase the data related to repeating tasks when syncing. (I understand the opensource aspect of the app, and I'm graceful of the work already being done, I'm not complaining or asking for work to be done, I'm just adding my experience report when syncing to an app that supports it :smile:  )

**Comment by timkrief** [comment]
I might have a look see if I can make a PR that at least preserves the repeating aspect when syncing (I added a task to my todo list :rofl:)

**Comment by raimund-schluessler** [comment]
> I might have a look see if I can make a PR that at least preserves the repeating aspect when syncing (I added a task to my todo list 🤣)

That would be very much appreciated. Repeating tasks is on the roadmap for Tasks v0.14.0, but I would like to update the right sidebar first and introduce tabs there https://github.com/nextcloud/tasks/issues/664, otherwise the sidebar will be to crowded with the additional elements for repeating tasks.

**Comment by billgross** [comment]
+1 (actually +1000) for repeating tasks.  Many people find this an essential feature, myself included.

That said, I agree with @raimund-schluessler that the sidebar needs to be updated, and probably ahead of repeating tasks for repeats to work well.

**Thanks to all of you who do development on this and on all things NextCloud.**  The entire project is awesome!! (No matter what else we wish it would do.)

**Comment by akrea** [comment]
Double that! Recurring tasks and an android app would be very helpful. Unfortunately I do not have time nor do I know the language to program such an app (although I would love to do that)...

**Comment by raimund-schluessler** [comment]
@akrea There are several android apps which sync via CalDAV and can be used in combination with Nextcloud Tasks. We will not develop another one here.

**Comment by Jerome2103** [comment]
I like it too

**Comment by karan** [comment]
Just to add a data point, here's what I've noticed with the tasks.org client (NC tasks over CalDAV):

* In tasks.org settings, enable `Let server schedule recurring tasks`. Create a recurring task that's added to your calendar on NC. The task will sync and show up on NC tasks UI. Now, no matter where you mark the task complete, the recurrence/sync doesn't work as I would expect it to. The entire task is marked done instead of just that one occurrence.
* In tasks.org settings, disable `Let server schedule recurring tasks`. Now, marking repeating tasks done (in the tasks.org Android app or the NC Tasks web UI) works as expected.

I'm not certain if/when the `Let server schedule recurring tasks` is needed for NC tasks.

**Comment by abaker** [comment]
Sorry, that setting is not properly explained. When toggled on it disables Tasks.org's recurring task logic. It is intended to be turned on for servers that handle recurrence, like Open-Xchange

**Comment by agentlibre** [comment]
> I am also in favor of repeating tasks. I wanted to add a few more thoughts.
> 
> Suppose I schedule a repeating task for every Monday. On Monday, when I look at my "Today" tasks, it should appear. If I mark it off as complete, it should disappear until next Monday, when it should reappear. I suspect there's wide consensus on that point.
> 
> If I don't do it on Monday, however, to my mind it's still outstanding, so on Tuesday it should still appear as an (over-)due task. There may be a few cases where people will want it to just disappear (e.g. if you didn't take out the garbage on Monday, there's no point in doing it on Tuesday), but those are probably few enough that just checking them off as done is sufficient. Perhaps an option is needed there, but I think it's probably low priority.
> 
> For tasks that are completed late, there's the question of when to reschedule them. In many cases, if it is scheduled for Monday, it should be rescheduled for next Monday regardless of when it has been completed. However, there are other cases where the interval is more important then the start date. Suppose I want to change the oil in my car every three months. I set up a task to start on January 1 and repeat every 3 months. Come April, though, I'm out of town and don't change the oil until April 15th. Some people will think the next oil change should be June 1, but others will want June 15 (because that's when the oil is 3 months old). Other scheduling apps handle this by allowing rescheduling to be on a date (1st of the month every 3rd months) or on an interval (every 3 months).
> 
> Anything that can be done for repeating tasks would be wonderful. The above is just to help raise the issues a developer may want to consider.
> If I complete the task on Tuesday, it should disappear until

I second the above comment. The idea is not to scroll a list of endless new tasks but to enable the monitoring of (recurring) tasks on a day-by-day or week-by-week.

A very important aspect not to be omitted in the development: recurring tasks should be made **compatible with Nextcloud Calendar and Nextcloud Deck**.

Thank you very much for everything you do for the community! Nextcloud is a wonderful ecosystem! 🙏🏻

**Comment by petesimonic** [comment]
Just look into the logic of https://tasks.org/ for recurrence handling. They provide ideal approach; and we don't going in the direction to reinvent anything better here in comments.

> > I am also in favor of repeating tasks. I wanted to add a few more thoughts.
> > Suppose I schedule a repeating task for every Monday. On Monday, when I look at my "Today" tasks, it should appear. If I mark it off as complete, it should disappear until next Monday, when it should reappear. I suspect there's wide consensus on that point.
> > If I don't do it on Monday, however, to my mind it's still outstanding, so on Tuesday it should still appear as an (over-)due task. There may be a few cases where people will want it to just disappear (e.g. if you didn't take out the garbage on Monday, there's no point in doing it on Tuesday), but those are probably few enough that just checking them off as done is sufficient. Perhaps an option is needed there, but I think it's probably low priority.
> > For tasks that are completed late, there's the question of when to reschedule them. In many cases, if it is scheduled for Monday, it should be rescheduled for next Monday regardless of when it has been completed. However, there are other cases where the interval is more important then the start date. Suppose I want to change the oil in my car every three months. I set up a task to start on January 1 and repeat every 3 months. Come April, though, I'm out of town and don't change the oil until April 15th. Some people will think the next oil change should be June 1, but others will want June 15 (because that's when the oil is 3 months old). Other scheduling apps handle this by allowing rescheduling to be on a date (1st of the month every 3rd months) or on an interval (every 3 months).
> > Anything that can be done for repeating tasks would be wonderful. The above is just to help raise the issues a developer may want to consider.
> > If I complete the task on Tuesday, it should disappear until
> 
> I second the above comment. The idea is not to scroll a list of endless new tasks but to enable the monitoring of (recurring) tasks on a day-by-day or week-by-week.
> 
> A very important aspect not to be omitted in the development: recurring tasks should be made **compatible with Nextcloud Calendar and Nextcloud Deck**.
> 
> Thank you very much for everything you do for the community! Nextcloud is a wonderful ecosystem! 🙏🏻

**Comment by Mela** [comment]
Yes, please add repeating/recurring tasks. I just switched from Toodledo to Nextcloud Tasks and was bummed out that this feature does not allready exist. 
I have quite some tasks I need to do on a weekly, monthly or yearly basis and I really liked the option to have them pop up when they are do without having to add a new task every time.

**Comment by petesimonic** [comment]
Most viable workaround I saw was adding "create a same task with due date in particular period" into the task manually and count it as done only if it is created.

P. S. Was really surprised Thunderbird don't understand *repeat in particular period *after done**.

> Yes, please add repeating/recurring tasks. I just switched from Toodledo to Nextcloud Tasks and was bummed out that this feature does not allready exist.

You can use it just as CalDAV server. Then recurrence will be managed by client.
> I have quite some tasks I need to do on a weekly, monthly or yearly basis and I really liked the option to have them pop up when they are do without having to add a new task every time.

**Comment by d-chell** [comment]
Wanted to second this feature request, agreeing with how agentlibre sees it

> The idea is not to scroll a list of endless new tasks but to enable the monitoring of (recurring) tasks on a day-by-day or week-by-week.
> A very important aspect not to be omitted in the development: recurring tasks should be made compatible with **Nextcloud Calendar** and **Nextcloud Deck.**

I'd love to have everything centralized in Nextcloud without integrating tasks.org, if possible.

Thanks to all the NC developers!

**Comment by meichthys** [comment]
If you're interested in this feature, contribute to the bounty here:
✅ https://www.bountysource.com/issues/40960486-feature-request-repeating-tasks
Apart from contributing code, this is probably the best way to speed up development on a feature request like this one. Support our open source developers! 🚀

**Comment by Extarys** [comment]
Taking a look at this if I can implement it with my limited time and knowledge :joy: 

How should completed reoccurring tasks looks like?
Right now:
![List of tasks](https://user-images.githubusercontent.com/11581433/132063777-cc41c4db-5781-4180-9283-1e32b01fcba1.png)

Completed tasks have the date/time at which they were completed. I never really used a Task/To-do list app before therefore I'm not sure how it should be managed. I would like some screenshot and comments about how it works elsewhere and what people would like.

0. My understanding is that Nextcloud server already have the capabilities for reoccurring events, as it is implemented in the Calendar app, therefore all the work needed can happen inside the Tasks app.
1. Should reoccurring tasks have their due date before the next one?
1.1. Assumption 1: A daily task max due date would be before the next day
1.2. Assumption 2: A failed daily task shouldn't allowed to be "completed", but marked as "failed"
2. I would like the implement a list of tasks with more options, like some kind of "habits" list, if 1.1 shouldn't be implemented in the current task system.
2.1 Some kind of graph/data history about reoccurring tasks could be dope
3. Contrary to the Calendar app, where there is a repeat tab in the info sidebar, I would put the reoccurring options inside the details tab if it doesn't make it feel too cluttered, although the parameters would need to be multi-line, not like everything is right now. I'm thinking about collapsing the settings related to this.

Right now:
![Task options](https://user-images.githubusercontent.com/11581433/132065326-085e3f5b-4586-44e1-9db3-e341e1ef2ea8.png)

I'll play with the UI and post some WIP eventually but I'd like to gather visual data about how people expect this to work and look.

**Comment by meichthys** [comment]
0. This is my understanding as well. I have reoccurring reminders being synced via iOS Reminders without too much trouble.

1. No, I don't think so. Generally most task managers do not relate the next task until the previous one is completed, so the task should never exist twice as incomplete. I think most task managers have no due date by default.

2. This sounds interesting but should probably be tabled until the repeating tasks are properly handled IMO.

3. I completely agree.

I look forward to following your progress and would be glad to test anything out if needed 👍

**Comment by crazybadger** [comment]
Thanks for looking at this.

My main use-case for repeating tasks is working within the GTD framework, so I have regular daily tasks that I complete and that recreate with a set end date.

For example, I have a Review Inbox task that I complete every day and it repeats the following day as I’ve set it to repeat daily.  I have a Review Weekly Actions task that I complete weekly and when I complete it create a new task 1 week in the future.

I’d also like to be able to create task for checking things on a more infrequent basis (e.g. updating SSL certificates or server packages) but when I complete them they repeat with a end date based on the repeat frequency I’d set on the task.

Hope that helps?

Adrian

**Comment by raimund-schluessler** [comment]
>    4. Contrary to the Calendar app, where there is a repeat tab in the info sidebar, I would put the reoccurring options inside the details tab if it doesn't make it feel too cluttered, although the parameters would need to be multi-line, not like everything is right now. I'm thinking about collapsing the settings related to this.

We will create a separate Tab for repetitions in the sidebar (there even is a placeholder already):
https://github.com/nextcloud/tasks/blob/aebdb4addb75bca254d0c3f247b793070bca77d8/src/views/AppSidebar.vue#L243-L250

The General / Info tab contains quite a lot of entries already, and with repetitions, it will be to cluttered. But that is a minor implementation detail not to worry to much about at the moment.

**Comment by spacekpe** [comment]
FTR Mozilla Thunderbird has full support for repeating tasks, so maybe it's user interface can serve as an example how to sort out corner cases?

For repeating tasks, it "instantiates" a new task for each repetition even if the previous one was not finished.
E.g. if the task is monthly "pay water bill" on every 1st day of month, a new unfinished task will show up even if the previous one is not marked as finished. So if the user did not confirm paying water bill six months in a row, the task list will display six unfinished tasks each with different due date in past - which reflects the reality of missing payments for six bills.

**Comment by spacekpe** [comment]
Possible recurrence rules can be pretty complicated, which is why all _complete_ recurrence UIs have lots of knobs as well. See https://datatracker.ietf.org/doc/html/rfc5545#section-3.3.10 for recurrence specification.

**Comment by timkrief** [comment]
If I can add to the conversation, I think the way the UI/UX of the tasks.org app works for repeating tasks is really good and could be a good inspiration.

**Comment by djibux** [comment]
> For repeating tasks, it "instantiates" a new task for each repetition even if the previous one was not finished.
> E.g. if the task is monthly "pay water bill" on every 1st day of month, a new unfinished task will show up even if the previous one is not marked as finished. So if the user did not confirm paying water bill six months in a row, the task list will display six unfinished tasks each with different due date in past - which reflects the reality of missing payments for six bills.

I believe one should be able to tell the app if it must repeat the task according to its due date or its completion date.
The second one is very useful.  For example if I need to get my car checked every year, or change the filter on my water filter jug every months, the app needs to instantiate a new task upon completion, not upon due date.  If I tend to procrastinate, or if I'm away on vacation, the two dates could be a few weeks apart.

The todo.txt "standard" implemented for instance in [Simpletask](https://f-droid.org/en/packages/nl.mpcjanssen.simpletask/) allows that.  Thunderbird does not as far as I know.

**Comment by meichthys** [comment]
@Extarys Have you made any progress on this, or is there anything we can do to help out?

**Comment by JackDrydenUK** [comment]
Just jumping in here to say I also could really use this feature!

I can't program but I'm more than happy to test anything or offer advice on UX design where helpful.

**Comment by Discostu36** [comment]
> For repeating tasks, it "instantiates" a new task for each repetition even if the previous one was not finished.
> E.g. if the task is monthly "pay water bill" on every 1st day of month, a new unfinished task will show up even if the previous one is not marked as finished. So if the user did not confirm paying water bill six months in a row, the task list will display six unfinished tasks each with different due date in past - which reflects the reality of missing payments for six bills.

This might be useful for paying bills but it is not for most recurring tasks that I have. I don't need to clean the bathroom two times in a row when I have skipped it once.

**Comment by alexghergh** [comment]
Well in this case we can agree there could possibly be an option (for a particular task) to change between the behaviour? I am not sure however how this can be stored in a database and used accordingly.

**Comment by djibux** [comment]
Well the feature I mentioned above is a simple solution and seems to be pretty standard: deciding on weather the new event should be created from the due date or the completion date.

I use both, they both make sense.

Here is a screenshot from [https://tasks.org/](https://tasks.org/):
![Screenshot_20211031-202310_Tasks](https://user-images.githubusercontent.com/6897338/139598451-c71cdd12-387b-4115-9d44-57778d840201.png)

**Comment by meichthys** [comment]
@djibux Agreed. They  both definitely have their place. Now let's argue about which one is the default 🤣
IMO, repeat based on completion date makes sense in most scenarios, and is how iOS Reminders handles it, but I'm ok either way.

**Comment by jpschewe** [comment]
I'm currently using tasks.org with Nextcloud and in the past used Toodledo. Both have the option to repeat based on due date or completion date; both default to repeat based on due date.

Using tasks.org I can have it handle the repeating appointments or delegate to the server. When this is implemented in Nextcloud Tasks should there be an option to specify if Nextcloud should handle the repeat or not? Or should Nextcloud always handle the repeats?

**Comment by Discostu36** [comment]
> Using tasks.org I can have it handle the repeating appointments or delegate to the server. When this is implemented in Nextcloud Tasks should there be an option to specify if Nextcloud should handle the repeat or not? Or should Nextcloud always handle the repeats?

At the moment, tasks.org marks the task completed, changes its due date and reopens it. As soon as Nextcloud can handle repeated tasks, tasks.org doesn't have to do that anymore.

**Comment by Eschguy** [comment]
Just adding my voice of support for this feature! Looks like it's been a long process!

**Comment by f0rkth1s** [comment]
Can't wait for this feature to be functioning! Thanks.

**Comment by trevholland** [comment]
Just chiming in that I would also love this feature!

**Comment by MartinConsultingServicesInc** [comment]
It strikes me that there are really two parts to this request:
1. Allow NextCloud to store and retrieve repeating tasks in the database, if only for sync purposes.
2. Present repeating tasks in the web-based UI.

I believe that we currently have neither, though it's been a while since I tested #1 as it used to trash my repeating tasks. Is that still the case, or does NextCloud successfully sync and retain task repetition data?

While we want both, of course, just getting #1 in would allow external applications (Tasks.org, OpenTasks, etc) to access and operate on task data in the Nextcloud, including repeating tasks. For many users, these apps are their *only* interface with Task management, so they don't care about the UI. In my case, I'd like to sync my phone-based task data with NextCloud so that it would get captured in my NC database backups for recovery purposes. I still want the UI, but it's a lesser priority.

So, while we want both, just having #1 in production would be beneficial to a lot of users, with #2 following. Thoughts?

**Comment by bpmartin20** [comment]
Sorry ... prior post regarding two parts that is now attributed to "ghost" was mine. Cleaned up my old github accounts and the attribution went away.

**Comment by meichthys** [comment]
I believe number 1 is already handled - at least in some capacity since I can sync repeating tasks from my iPhone to nextcloud and back to my Mac without any serious issues.

**Comment by meichthys** [comment]
The problem comes when I try to interact with a repeating task using the web interface. At that point it strips off the repeating task and o think I’ve seen it duplicate the task. It’s a mess. Don’t try to use the web I terrace with repeating tasks!

**Comment by BaderSZ** [comment]
> I believe number 1 is already handled - at least in some capacity since I can sync repeating tasks from my iPhone to nextcloud and back to my Mac without any serious issues. (Both the Mac and iPhone show the tasks as repeating tasks.)

The issue that I can think of is that a proper implementation (with [RRULE](https://icalendar.org/iCalendar-RFC-5545/3-8-5-3-recurrence-rule.html)) might break some clients, although I do not care at this point. The main issue for me would be figuring out how to properly integrate this to the UI.

Question: Should we implement calendar-js first?

Given that the issue has been open since 2017, it seems to be a non-priority to the devs. So I'll take a look over the weekend. No promises.

**Comment by raimund-schluessler** [comment]
> Question: Should we implement calendar-js first?

I think that would be the best approach, as calendar-js can handle repetitions already. But some adjustments to make it work properly for Tasks are necessary for sure.

> Given that the issue has been open since 2017, it seems to be a non-priority to the devs. So I'll take a look over the weekend. No promises.

Well, it's a very complex feature, UI as well as logic wise. And given limited (free)time that I can dedicate to this app, there are lower hanging fruits that are easier to implement and helpful as well.

**Comment by BaderSZ** [comment]
> > Given that the issue has been open since 2017, it seems to be a non-priority to the devs. So I'll take a look over the weekend. No promises.
> 
> Well, it's a very complex feature, UI as well as logic wise. And given limited (free)time that I can dedicate to this app, there are lower hanging fruits that are easier to implement and helpful as well.

I definitely understand. And as we both might say in german, das war nicht böse gemeint. :)

**Comment by raimund-schluessler** [comment]
> I definitely understand. And as we both might say in german, das war nicht böse gemeint. :)

All good, no offense taken :slightly_smiling_face:

**Comment by RayBB** [comment]
@raimund-schluessler 
I think the biggest problem right now is that when you check off a task that has an RRULE NC (nextcloud) not only ignores that but also removes it.

The smallest step I can think of to make an improvement here is to just make it so that if `RRULE` is present move the `DTSTMP` forward appropriately. We can do this without making an HTML/CSS changes at all.

The major downside here is that if a user checks off a task that is recurring the task will reappear without any clear reason. Personally, I think this is a reasonable tradeoff because:
1. Nobody who just uses nextcloud would have an RRULE set
2. They would have to be using some other client and are probably aware of the problem of recurring tasks breaking in NC anyway.
3. They can still delete the task if they choose

Once that is complete NC tasks will stop wiping out recurring tasks and then someone can work on the UI side of things to indicate that a task is recurring and set recurrence.

**Do you think this is a reasonable path forward?** If so, I am willing to dig in and try to get RRULE acknowledged.

If anyone would like to work on this with me, or just share ideas feel free to Tweet me https://twitter.com/RayScript I'm pretty comfortable in the JS world but this will be my first time touching a NC app (but I already have the dev env setup locally!)
cc: @BaderSZ

**Comment by RayBB** [comment]
@raimund-schluessler 
For the UI side of things I think we can reuse the [Repeat component](https://github.com/nextcloud/calendar/tree/v3.3.2/src/components/Editor/Repeat) from nextcloud/calendar. It looks like [this](https://user-images.githubusercontent.com/921217/172036400-159d350b-b5f7-4da9-9b13-af1c79a9b098.mp4). 

To do that, I think we might need to tweak it to be a bit more generic. Then move it to https://github.com/nextcloud/nextcloud-vue, which seems to be the place to share Vue components in nextcloud.

Does this sound like the correct approach? If so, should I open an issue in nextcloud-vue, nextcloud/calendar, or both?

**Comment by BaderSZ** [comment]
> Does this sound like the correct approach? If so, should I open an issue in nextcloud-vue, nextcloud/calendar, or both?

I'm presuming this wasn't included in nextcloud-vue because NC Calendar was the only App using it. It depends on what we want to do, or specifically, what the nextcloud team prefers: making a new repeat component more fitting for tasks, or reusing this one.

I'd prefer to re-use, but it's not really my call.

**Comment by Horuyuki** [comment]
So is it possible to set a recurring task? On an iphone reminder we can do that and the task actually repeats. Is this done by the iphone reminder app or by the task app?

**Comment by meichthys** [comment]
Yes, the repeating tasks do work fine in iOS, you just can't use the Nextcloud web interface to create/modify them otherwise the repeating part of the task will be corrupted.

**Comment by Horuyuki** [comment]
> Yes, the repeating tasks do work fine in iOS, you just can't use the Nextcloud web interface to create/modify them otherwise the repeating part of the task will be corrupted.

That is why, after I marked the repeat tasks off on nextcloud they are no longer working in iOS...

Will the feature be native in nextcloud soon? I think github says it will be in the next version?

**Comment by meichthys** [comment]
> That is why, after I marked the repeat tasks off on nextcloud they are no longer working in iOS...

That's correct. The web interface breaks repeating tasks, so it's best not to use it for those.
Where did you see that it will be in the next version?

**Comment by raimund-schluessler** [comment]
Yes, it is correct that if I find enough time to work on a new major feature of the Tasks app, it will first be repeating tasks. However, keeping the app usable for new Nextcloud versions and adjusting/fixing minor issues (aka maintaining the code) already takes most of the time I can dedicate to the app. That's why I can't give any timeline for the implementation of this feature, as much as I would like to.

**Comment by Horuyuki** [comment]
Hey, I have to thank you devs for making private data hosting services like tasks possible.
I give you my best wishes on reaching this next big milestone!

**Comment by jc97** [comment]
What about making repeating tasks in the web interface read-only until they are supported by the web interface?

**Comment by petesimonic** [comment]
Let me ask kind of a stupid question. What will be with repeating task in a NC server, which were repeating for years after the feature will go live? I mean the ones created by Tasks (the Android app) primarily, and also those created with Thunderbird (though it seems like Tb manages it fully on its end). Will they become repeating in NC interfaces too, or what?
> So is it possible to set a recurring task? On an iphone reminder we can do that and the task actually repeats. Is this done by the iphone reminder app or by the task app?

**Comment by BaderSZ** [comment]
> Let me ask kind of a stupid question. What will be with repeating task in a NC server, which were repeating for years after the feature will go live? I mean the ones created by Tasks (the Android app) primarily, and also those created with Thunderbird (though it seems like Tb manages it fully on its end). Will they become repeating in NC interfaces too, or what?

Presuming the other apps/programs/clients work as prescribed by RFC 5545, that shouldn't be an issue. So far, I think you'd only have to configure the tasks.org app to use server-side recurrence instead of the app. It's still a WIP and I am yet to test it, but that is my understanding

**Comment by meichthys** [comment]
It would be interesting to see how iOS handles this. I do not see any option in the iOS account settings to specify service side reoccurrence although there may be some kind of auto-configuration for this that I am not aware of.

**Comment by BaderSZ** [comment]
> It will be interesting to see how iOS handles this. I do not see any option in the iOS account settings to specify service side reoccurrence although there may be some kind of auto-configuration for this that I am not aware of.

I would hope that Apple follows the standard, but given that it's not very thoroughly defined in some aspects, I expect some diversions. Could you send me a link/source or even example ICS file with recurring tasks, if you could export them on your iOS device? You should find my email in my profile.

**Comment by meichthys** [comment]
@BaderSZ I don't think there is an easy way to export an `.ics` file from iOS.
Here's the `.ics` file after I create it on iOS, then sync to nextcloud, then export from nextcloud:
[ics_created_in_ios_exported_from_nextcloud.zip](https://github.com/nextcloud/tasks/files/9066551/ics_created_in_ios_exported_from_nextcloud.zip)

There is probably a way to export the `.ics` file from MacOS which should be similar to iOS. I'll try to provide that as well.

**Comment by meichthys** [comment]
It looks like Apple removed the export functionality in the Reminders app. Perhaps someone with an older version can export an `.ics` file that includes a repeating task.

**Comment by meichthys** [comment]
@BaderSZ and @RayBB,
Is there anything else we can provide to help get things moving on this issue? I'm not well versed on php or vue, but I'd like to help where I can to get this working.

**Comment by BaderSZ** [comment]
> @BaderSZ and @RayBB, Is there anything else we can provide to help get things moving on this issue? I'm not well versed on php or vue, but I'd like to help where I can to get this working.

I just finished my exams, so my little 'break' is over. :)

 php isn't specifically the issue, but a functional 'calendarObjectInstance'-style (JS/TS) implementation is. I'll try to keep up with progress and report in my repo soon.

**Comment by meichthys** [comment]
@BaderSZ Sounds great. We're rooting for you! If there's any we can do to help, let us know!

**Comment by warent** [comment]
@BaderSZ I'm a full stack engineer with a focus in the frontend. Would love this for my nextcloud instance. Let me know how I can help get this across the finish line

**Comment by meichthys** [comment]
@BaderSZ, Any update on this? I don't see any changes [here](https://github.com/nextcloud/tasks/compare/master...BaderSZ:nextcloud_tasks:master), but I don't want to assume that is your only repository for the changes you've been working on.

**Comment by BaderSZ** [comment]
Hey @meichthys ,

you're right, I've had to take the 3  weeks off for family reasons, and I do apologise for that. Would you send me an email so we can discuss timeline and goals?

**Comment by thrdroom** [comment]
This feature request is being around since 2017, and its still not implemented. Yet Nextcloud is implementing more more features which also get implemented only halfway. I'm so done with Nextcloud as a project.

**Comment by Extarys** [comment]
@thrdroom I understand your feelings, but if you are using the community version, features like that are implemented by the community (mostly).

I know it _sometimes _sucks_, but you can either contribute yourself by code, or by making detailed issues, paying for NextCloud Enterprise (which provide more support and helps make features like this) or hire a dev that will implement the features you want. I, too, would like a thousand features with no bugs, but sadly it doesn't work like that :/

**Comment by nkakouros** [comment]
A genuine (and not sarcastic) question I have is how can enterprise users use Nextcloud without repeating tasks, reminder configuration and other PIM suite features. Are they all really so low in requirements?

**Comment by sunjam** [comment]
You can use clients for the repetition of tasks and calendars on mobile and
desktop.

**Comment by meichthys** [comment]
> A genuine (and not sarcastic) question I have is how can enterprise users use Nextcloud without repeating tasks, reminder configuration and other PIM suite features. Are they all really so low in requirements?

We can use nextcloud for repeating tasks as long as you don't use the nextcloud web client. 
we use Thunderbird which works reasonably well. We will definitely migrate to using the web interface when repeating tasks are implemented!

**Comment by BHuck74** [comment]
Dear all,
I've migrated from Synology calendar to Nextcloud for some reasons. Nextcloud is just a great platform. However, I'm just stunned that NC calendar is not able to manage recurring tasks!!! I will of course stay on NC but  need to find an alternative now to manage all my recurring tasks. 
This would be of course a great feature to be added!

**Comment by sunjam** [comment]
You'll need to manage recurring tasks using a third party client.  I can
think pf Thunderbird and Super Productivity plus various mobile apps.

**Comment by SebastianBerlin** [comment]
I totally missing that necessary function!

**Comment by buttercheetah** [comment]
I would also greatly appreciate this feature being added

**Comment by sunjam** [comment]
Clients = desktop and mobile clients. So, not the server app or webui

**Comment by meichthys** [comment]
@f1d094 Yes, all versions of NC support this - but you must be using a client (not the nextcloud web interface) to interact with the tasks lists.
See here for some clients that work with Nextcloud tasks: https://github.com/nextcloud/tasks/#apps-which-sync-with-nextcloud-tasks-using-caldav

**Comment by DriedPlumber** [comment]
still needed in 2023 :)

**Comment by link-leocorde** [comment]
Tell me where to transfer funds to support bringing recurring tasks into Nextcloud, natively.

**Comment by sunjam** [comment]
They are not accepting donations.

**Comment by meichthys** [comment]
@BaderSZ and @warent Do either of you still have this issue on your schedules?
I'm just trying to offer help where I can even though I don't yet have the expertise to implement this myself.

**Comment by timunrau** [comment]
Hi all, add me to the long list of people waiting for this feature. I would like to migrate away from Microsoft ToDo to NextCloud tasks but lack of reoccurring tasks is making this impossible. There are no good desktop clients for tasks on Linux. I would like to use the web UI for NextCloud tasks as it is actually pretty good.

**Comment by meichthys** [comment]
@timunrau You may want to check out Kalendar. It's quite nice. 
Thunderbird and Evolution are also decent.

**Comment by timunrau** [comment]
Thanks @meichthys I'll check it out!

**Comment by bpmartin20** [comment]
There are a lot of people that want this. Maybe we need to put a bounty 
on it, to entice someone to spend a couple of their weekends on it. I've 
never participated in a bounty system. Is there one that the Nextcloud 
devs and would-be devs prefer?

                         -Brian M.

**Comment by meichthys** [comment]
@bpmartin20 Bounties used to be done via [Bountysource](https://help.nextcloud.com/t/dont-use-bountysource-anymore/84943/52), but no longer. AFAIK there has been no official Nextcloud statement on where to issue bounties. 

There is however a ['freelancing' section in the forum](https://help.nextcloud.com/c/nextcloud-freelancing/) that may be our best location for such a 'bounty'.

**Comment by bernhard1701** [comment]
I also would love to have the recurring tasks feature

**Comment by WebworkrNet** [comment]
This missing, in my eyes, essential function is currently preventing me from switching to Nextcloud Tasks. :-(

**Comment by AbelLykens** [comment]
I can recommend [jtx](https://jtx.techbee.at/sync-with-davx5) for android, but still would like native support. Too bad there are no bounties anymore.

**Comment by AbelLykens** [comment]
Correct. Some more background:
When you set up a recurring task (for example in jtx), it uses the `RRULE` property it seems:

```
BEGIN:VTODO
DTSTAMP:20230510T135913Z
UID:99999999-0000-1111-2222-33333333333333
SEQUENCE:12
CREATED:20230510T135521Z
LAST-MODIFIED:20230510T135840Z
SUMMARY:Water the plants
CATEGORIES:House
DTSTART:20230512T070000
RRULE:FREQ=DAILY;COUNT=-1;INTERVAL=7
PRIORITY:0
DUE:20230513T140000
END:VTODO
```

These recursion properties (`RRULE`, `EXDATE` et cetera) are properly stored by the dav backend of nextcloud, so that's good.
As @meichthys indicates, the frontend is missing features here. Which is correct: if I check-complete a task in the frontend, it completes all the tasks in the series (because the front end does not understand recursion).

Implementing this will not be easy, I can think of a lot of exceptions, UI stuff you'd need to implement. But it would be appreciated a lot.

**Comment by sunkup** [comment]
Miss this feature too. It prevents me from moving to nextcloud. Completely new to nextcloud app development, but I will play around a bit. Maybe something useful comes out of it.

**Comment by sunkup** [comment]
@meichthys That's very kind of you, but money is not an issue. No need to donate. I am doing this for fun and because I need the feature myself. The limiting factor is - as so often - time. I took a bigger holiday break, but as things return to everyday life I hope to be able to work a few hours per week on this again. 

As of right now, the core functionlity stands, but creating the sidebar UI properly is much more work than I anticipated (even when re-using code from nextcloud/calendar). 

About helping out: Once it's ready and I wrote some unit tests, I would love to have a few people test it thoroughly! Since this is my first time working on a nextcloud app, it is very likely there will be bugs which need be addressed before merging.

**Comment by llzzrrdd** [comment]
I know this is probably not the right place to comment on Nextcloud's choices, but really, collaborative platform, with endless issues for years now with the #1 collaborative tool people use.... calendar with tasks? Really? I am out, I uninstalled Nextcloud and have a Baikal server and other open source servers that work independently from each other, stable for many years.... happy new year!

**Comment by sunkup** [comment]
> I don't intend to rush things, but @sunkup do you think there is any chance this could land in v0.16? Thanks again for your work on this!

No, I don't think so. My priorities lie elsewhere in life right now. I really just work on this sporadically. While I will continue to work on this, I can give no ETA whatsoever. 

On that note - I don't mind someone else taking over or helping me out where possible :)

**Comment by XotEmBotZ** [comment]
What is the status of this feature?

**Comment by PolynomialDivision** [comment]
> What is the status of this feature?

As you can see here: https://github.com/orgs/nextcloud/projects/61/views/4?filterQuery=-status%3A%22%F0%9F%A7%AD+Planning+evaluation%22+repeating

It is "in progres" however, I have no idea what "no planned" group means.

**Comment by Felitendo** [comment]
This has already been implemented for the Nexcloud Calendar App. Maybe someone from the Nextcloud GmbH who was able to implement this on the Calendar App could also take a look at the Tasks App?

**Comment by emes81** [comment]
https://github.com/nextcloud/tasks/pull/2318#issuecomment-2619453477

"we (Nextcloud GmbH) are not working on this right now."

It looks like implementation of recurring tasks will have to wait until a community member is able to implement it. A real shame. It's a core feature of a tasks app. And I don't want to use a proprietary tasks app.

**Comment by adamchap1** [comment]
And then the usual chorus of "use FOSS" and Nextcloud is amazing breaks out. This is the absolute perfect example why people use proprietary software - FOSS developers prioritise dogma above practicality.

**Comment by timkrief** [comment]
Don't be so negative you all. Honestly I would be glad to work on this but it makes no sense financially, maybe one day I will have the means to just spend weeks on open source development.

**Comment by oktayacikalin** [comment]
> and we can work out a way of funding that.

Yes, perhaps we can do a kickstarter? 🙂

**Comment by RayBB** [comment]
I'll commit $50 to whoever creates a PR for this that gets merged into the main repo. Valid over year from now. Will send via PayPal or Wise.

**Comment by k8ieone** [comment]
I'd be able to add ~15 USD to that

**Comment by bpmartin20** [comment]
Ditto for $50 USD. We're up to $115 USD in 7 minutes, but we should 
really use one of the established bounty boards for this. Anyone know of 
a reputable one?

**Comment by RayBB** [comment]
Generally, as far as I know the bounty boards require open in by repo to link to issues. Which also includes the maintainers agreeing to actually review such a PR.

**Comment by djagoo** [comment]
I would add $100 USD to that.

**Comment by emes81** [comment]
> Generally, as far as I know the bounty boards require open in by repo to link to issues. Which also includes the maintainers agreeing to actually review such a PR.

What's the proper way to approach the maintainers to get their agreement to this? And is there a preferred way to formalise/collect the bounty for this feature, so that anyone working on the feature will know that they will get what was promised?

**Comment by beangeli** [comment]
> > Generally, as far as I know the bounty boards require open in by repo to link to issues. Which also includes the maintainers agreeing to actually review such a PR.
> 
> What's the proper way to approach the maintainers to get their agreement to this? And is there a preferred way to formalise/collect the bounty for this feature, so that anyone working on the feature will know that they will get what was promised?

Also curious about how to formalize and bring attention to this. Count me in for $500 once we figure that out.

**Comment by DJCrashdummy** [comment]
i would say it must respect [RFC 5546](https://datatracker.ietf.org/doc/html/rfc5546) and [RFC 5545](https://datatracker.ietf.org/doc/html/rfc5545)! because if it is designed to (only) work in sync with client X or Y, i'm sure there are a bunch of people who won't be happy and hence not release the bounty as the issue is not solved for them.

yes, i'm aware that this may cause some clients to still not work in sync with Nextcloud Tasks, but then it is on their users/developers/community to make their client comply with the standard.

**Comment by i2h3** [comment]
> Also curious about how to formalize and bring attention to this.

It already is known to be most desired feature based on community votes on GitHub. Personally, it is my biggest feature wish for Nextcloud, too.

**Comment by bpmartin20** [comment]
On 3/4/25 05:01, DJCrashdummy wrote:
>
> i would say it must respect RFC 5546 
> <https://datatracker.ietf.org/doc/html/rfc5546> and RFC 5545 
> <https://datatracker.ietf.org/doc/html/rfc5545>! because if it is 
> designed to (only) work in sync with client X or Y, i'm sure there are 
> a bunch of people who won't be happy and hence not release the bounty 
> as the issue is not solved for them.
>
> yes, i'm aware that this may cause some clients to still not work in 
> sync with Nextcloud Tasks, but then it is on their 
> users/developers/community to make their client comply with the standard.
>
I agree.

Furthermore, it seems to me that the Calendar folks at NC are very 
standards focused, so I think a non-standard solution would be a 
non-starter for them.

**Comment by jancborchardt** [comment]
Hi everyone, just to give a little info as there seems to be an expectation for Nextcloud GmbH to do something about this:
- We do discuss this regularly in planning for Groupware and Office, but for now have other things to focus on – especially regarding stability and performance, which is really relevant for everyone
- Thanks a bunch to @sunkup for his pull request! :) As written in the "list of what is (not) working", this is proof-of-concept and not in a mergable state and would still require significant work: https://github.com/nextcloud/tasks/pull/2318#issuecomment-2306802419
- If you are an organization who would like to see this implemented (and since there has been a lot of talk about bounties lately), we recommend requesting it over in the "Nextcloud freelancing" category of our forum: https://help.nextcloud.com/c/nextcloud-freelancing/48

Hope this explains the situation. Just to clarify: It’s not entirely off the table that Nextcloud GmbH would act on this, but stability and performance are (as you most likely know) big points we often get feedback about and need to focus on. And the awesome thing is that we are also an open source software community where everyone can contribute. :)

**Comment by bpmartin20** [comment]
> there seems to be an expectation for Nextcloud GmbH to do something about this

I think everyone understands that Nextcloud GmbH has other priorities that outweigh this one. I trust your judgement.

There is one thing that would help, though, that _**only**_ Nextcloud GmbH can do. As the owner of the repository, only Nextcloud GmbH can set up a bounty system. With a bounty system we disparate individuals scattered around the world could establish a proper bounty for this feature. Furthermore, anyone else could do the same for any other high-interest enhancement. Please, give us the tool to hire people to do work you don't have the resources for.

The Nextcloud freelancing forum is good, but it really only helps in situations where a single entity wants to pay the entire bill. The existing Nextcloud bug bounty system only applies to security. We really don't have a place where individuals and small organizations can pool their resources to encourage someone to program a much-desired enhancement.

Thank you, @jancborchardt for speaking up on this topic and letting us know that Nextcloud GmbH is listening.

**Comment by bcutter** [comment]
So what’s the status - how can users without dev skills support (e. g. donate) **in detail right now**?

After so many financial support commitments here it looks like we just shifted that collection of support to another space - from GitHub to the forums.

But where's the actual progress. **Where's the funding/collection tool, where's the pot.**

Time flies, we have 2026 now and it is absolutely hilarious to read every new NC Hub release blog post with all those great fancy features and AI here and AI there - while on the other hand there are much requested features like this here, open for 9 freaking years now.

**Comment by djagoo** [comment]
> Time flies, we have 2026 now and it is absolutely hilarious to read every new NC Hub release blog post with all those great fancy features and AI here and AI there - while on the other hand there are much requested features like this here, open for 9 freaking years now.

@bcutter absolutely nailed it! I think this project is doomed to fail because basic features are missing but with every new version more bloat get's added instead.

We started to migrate our customers to opencloud servers and even our customers in the federal government are happy with it (german federal government institutions insisted on nextcloud in the past). Opencloud uses Radicale as caldav/carddav backend. In the past we thought that the missing frontend-apps i.e. for calendar and contacts are a major drawback when using opencloud but there are good webmailers adding those functions back. On the other hand the frontend apps for contacts, calendar, tasks... are seemingly not beeing maintained anymore by Nextcloud GmbH or at least they are no priority for them

**Comment by qdii** [comment]
@djagoo not to derail, but a quick look at the github discussions doesn't make me trust that Opencloud is treating its open-source community correctly (e.g. https://github.com/orgs/opencloud-eu/discussions/2020). I'll defer my decision to migrate until I have more evidence that open-source is more than just a commercial argument to them.

**Comment by djagoo** [comment]
@qdii absolutely fair point. Out decision to migrate our customers was due to the fact, that they need i.e. this feature (repeating tasks) and I can't promise to them that this will come after years of waiting (not the only point but one of them). So far the matrix community of opencloud seems to be great and helpful and I have high hopes because of all the developers from owncloud who joined the team. Yes, it may be risky but so far it "feels good" for us. And as @bcutter statet, with every new release where a new ai feature / app is presented I am more disappointed in nextcloud. But yeah, time will tell.

**Comment by i2h3** [comment]
For anyone reading along: the web user interface of Nextcloud Tasks does not offer this but aniOS device set up for synchronization apparently still manages repeating tasks correctly. At least to what I have tested on my iPhone.

**Comment by DJCrashdummy** [comment]
@i2h3 well... (although not clearly stated in the title or OP) this issue was **never about syncing** repeating tasks, which always worked especially if you don't edit them via Nextclouds web-GUI.
if they still do get messed up when using Nextcloud just for syncing your tasks, it's very likely an issue of a least one of the client**s** you are using, as they might use a different format (resp. at least one of them certainly doesn't support the official standards - mentioned in https://github.com/nextcloud/tasks/issues/34#issuecomment-2697491925) and hence they seem to mess up the repeating tasks for each other (at the latest after editing them).

long story, short:
you can see the web-GUI just as a client like any other... and this issue is about adding support for **editing** repeating tasks via web-GUI.

**Comment by underclockeddev** [comment]
woot just in time

**Comment by bpmartin20** [comment]
So, do we have any idea when this will go into general release? I don't see a roadmap for Tasks. Is there a target date, or a general release schedule, or the like?

**Comment by devn1x** [comment]
To make it more visible: For anyone who wants to appreciate @schulm 's effort can find his PayPal here:
 https://github.com/nextcloud/tasks/pull/3021#issuecomment-4063667495

**Comment by dbmtrde** [comment]
Thank you very much for making this happen! 🫶
I also made a small donation to show my appreciation for your efforts. @schulm
